Update generated headers and clang format.

qihqi opened the pull request:
title:

Test plan: CI

iseeyuan commented on test/cpp/jit/test_save_load.cpp:
```diff
@@ -13,6 +15,20 @@
 
+namespace {
+
+static Module roundtrip(const Module& m) {
```
```suggestion
static Module roundtripThroughMobile(const Module& m) {
```

iseeyuan commented on test/cpp/jit/test_save_load.cpp:
```diff
@@ -13,6 +15,20 @@
+  std::vector<IValue> constants;
+  jitModuleToPythonCodeAndConstants(m, &files, &constants);
+  CompilationOptions options;
+  mobile::Module mobilem = jitModuleToMobile(m, options);
```
Could we also save the extra files to mobile module to have a more complete roundtrip?

qihqi replied:
extra files is orthogonal. 
The full roundtrip is
jit::Module -> (ivalue, source, constant) -> flatbuffer -> (ivalue, source, constant) -> jit::Module.
This test only tests first first and last step:
jit::Module -> (ivalue, source, constant)->jit::Module.

A test with extrafiles would be better aimed at testing torch::jit::load when it handles flatbuffer.

gmagogsfm commented on torch/csrc/jit/serialization/import.h:
```diff
@@ -98,5 +98,11 @@ TORCH_API Module load(
     c10::optional<c10::Device> device,
     ExtraFilesMap& extra_files);
 
+TORCH_API Module jitModuleFromSourceAndConstants(
```
Why do we need to emphasize it is JIT module when this is in torch::jit?

gmagogsfm commented on torch/jit/_serialization.py:
```diff
@@ -182,3 +182,72 @@ def validate_map_location(map_location=None):
+    try:
+        import torch._C_flatbuffer as ff
+    except ImportError:
+        print("Please include //caffe2:_C_flatbuffer as dependency.")
```
Is it possible for FlatBuffer dependency to be missing in OSS? If so, then we could probably have a dedicated error message for OSS

gmagogsfm reviewed (commented):
Awesome, thanks for working out a solution so quickly.

qihqi commented on test/cpp/jit/test_flatbuffer.cpp:
```diff
@@ -1135,5 +1135,91 @@ TEST(FlatbufferTest, OperatorTest2) { // NOLINT (use =delete in gtest)
+  mobile::Module mobilem = loader.parseModule(flatbuffer_module);
+  ExtraFilesMap files;
+  std::vector<IValue> constants;
+  loader.extractJitSourceAndConstants(&files, &constants);
```
yes.

qihqi commented:
@gmagogsfm  @iseeyuan as suggested I split the diff to make the generated header / clangformat change first. I'll publish another PR with the JIT source stuff.

qihqi commented on tools/linter/clang_format_all.py:
```diff
@@ -39,6 +39,8 @@ def get_allowlisted_files() -> Set[str]:
     for dir in CLANG_FORMAT_ALLOWLIST:
         for root, dirnames, filenames in os.walk(dir):
             for filename in filenames:
+                if filename.endswith('mobile_bytecode_generated.h'):
```
done.

iseeyuan reviewed (approved):
LGTM. Please make sure the size with JIT module would be on par with the pickle format, where constants are shared and the overhead of saving JIT module is small.

qihqi committed:
regenerate flatbuffer header (#73810)

Summary:
Update flatbuffer generated header and add it to ignore for clang format

Pull Request resolved: https://github.com/pytorch/pytorch/pull/73810

Test Plan: CI

Reviewed By: iseeyuan

Differential Revision: D34652217

Pulled By: qihqi

fbshipit-source-id: 9a7daac921da7bae1bdd761942c2489c6902453a
